components/engine_traits/src/flush.rs

```diff
@@ -103,12 +103,18 @@ impl SstApplyState {
 #[derive(Debug)]
 pub struct FlushState {
     applied_index: AtomicU64,
+
+    // This is only used for flush before server stop.
+    // It provides a direct path for flush progress by letting raftstore directly know the current
+    // flush progress.
+    flushed_index: [AtomicU64; DATA_CFS_LEN],
```

Comment:
If it's optional, it's better to be Option<[AtomicU64; DATA_CFS_LEN]>

Reply:
It's not optional.